sql/api/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseParser.g4

```diff
@@ -144,6 +148,22 @@ declareHandlerStatement
     : DECLARE (CONTINUE | EXIT) HANDLER FOR conditionValues (beginEndCompoundBlock | statement | setStatementInsideSqlScript)
     ;
 
+declareCursorStatement
+    : DECLARE strictIdentifier (ASENSITIVE | INSENSITIVE)? CURSOR FOR query (FOR READ ONLY)?
```

Comment:
not a big deal but why reserved keywords are not allowed here?

Reply:
That's an oversight... fixing